### DISCUSSION TITLE
Sonar Transducer needs an additional option "self-centered"

### DISCUSSION THREAD

**Discussion by zhu-rengong** [FakeFishGames/Barotrauma#6540] [Ideas and suggestions]
Sonar Transducer needs an additional option "self-centered", because the center of the sonar is always at the center of submarine, so in many cases we can't get required parameters.
Or provide an offset attribute relative to the center of submarine for nav terminal to calculate the center position of the sonar.

For example, easily implement operating the laser turret by navigation terminal, look this gif:
<img src="https://thumbnail0.baidupcs.com/thumbnail/666c0f5b8m44ccd3061d9caaf3717392?fid=2268879635-250528-187142507454307&time=1630512000&rt=sh&sign=FDTAER-DCb740ccc5511e5e8fedcff06b081203-X1pFFCwJQPEGFxsURWLzoAoWIRs%3D&expires=8h&chkv=0&chkbd=0&chkpc=&dp-logid=9118000297434971476&dp-callid=0&file_type=0&size=c710_u400&quality=100&vuk=-&ft=video"/>

Furthermore, for a large submarine with a length of 50 meters, it may require two sonar transducers.

**Comment by ubertpendragon** [comment]
I agree, but it's also very simple to work around this

Right now, you can use a shuttle with only a transducer and send the signal via wifi
It allows you to easily center the position of the sonar, and you can even swap channels for very big subs to center on different spots
![image](https://user-images.githubusercontent.com/73229309/131730252-5f6f4105-34b5-4c4d-ad15-2aa6896f31cf.png)


I assume you want to use a "sonar" turret, you simply need to dock a shuttle with transducer centered on the gun

**Reply by zhu-rengong** [reply]
Thank you very much for the solution!
